File: Data Structures & Algorithms/three-integer-sum/submission-1.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        set<vector<int>> s;
        for (int i = 0; i < nums.size() - 2; ++i) {
            int target = nums[i];
            int left = i + 1, right = nums.size() - 1;
            while (left < right) {
                if (nums[left] + nums[right] < -target) ++left;
                else if (nums[left] + nums[right] > -target) --right;
                else {
                    s.insert({target, nums[left], nums[right]});
                    ++left;
                }
            }
        }
        vector<vector<int>> res(s.begin(), s.end());
        return res;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_to_index = {}
        for i in range(len(nums)):
            if num_to_index.get(target - nums[i]) is not None:
                return [num_to_index[target - nums[i]], i];
            num_to_index[nums[i]] = i